src/Nancy/Json/JavaScriptSerializer.cs

```diff
@@ -29,459 +29,49 @@
-        JavaScriptTypeResolver _typeResolver;
-        bool _iso8601DateFormat;
+        bool retainCasing;
 
-#if NET_3_5
-        internal static readonly JavaScriptSerializer DefaultSerializer = new JavaScriptSerializer(null, false, 2097152, 100);
-
-        public JavaScriptSerializer()
-            : this(null, false, 2097152, 100)
-        {
-        }
+        readonly NancySerializationStrategy serializerStrategy;
 
-        public JavaScriptSerializer(JavaScriptTypeResolver resolver)
-            : this(resolver, false, 2097152, 100)
-        {
-        }
-#else
-        internal static readonly JavaScriptSerializer DefaultSerializer = new JavaScriptSerializer(null, false, 102400, 100, false, true, null, null);
+        internal static readonly JavaScriptSerializer DefaultSerializer = new JavaScriptSerializer(false, JsonConfiguration.Default);
 
         public JavaScriptSerializer()
-            : this(null, false, 102400, 100, false, true, null, null)
+            : this(false,JsonConfiguration.Default)
         {
         }
 
-        public JavaScriptSerializer(JavaScriptTypeResolver resolver)
-            : this(resolver, false, 102400, 100, false, true, null, null)
+        public JavaScriptSerializer(bool registerConverters, JsonConfiguration configuration)
         {
-        }
-#endif
-        public JavaScriptSerializer(JavaScriptTypeResolver resolver, bool registerConverters, int maxJsonLength, int recursionLimit, bool retainCasing, bool iso8601DateFormat, IEnumerable<JavaScriptConverter> converters, IEnumerable<JavaScriptPrimitiveConverter> primitiveConverters)
-        {
-            _typeResolver = resolver;
-            _maxJsonLength = maxJsonLength;
-            _recursionLimit = recursionLimit;
+            this.serializerStrategy = new NancySerializationStrategy(configuration.RetainCasing);
 
-            this.RetainCasing = retainCasing;
-
-            _iso8601DateFormat = iso8601DateFormat;
+            this.RetainCasing = configuration.RetainCasing;
 
             if (registerConverters)
-                RegisterConverters(converters, primitiveConverters);
-        }
-
-
-        public int MaxJsonLength
-        {
-            get
-            {
-                return _maxJsonLength;
-            }
-            set
-            {
-                _maxJsonLength = value;
-            }
-        }
-
-        public int RecursionLimit
-        {
-            get
-            {
-                return _recursionLimit;
-            }
-            set
-            {
-                _recursionLimit = value;
-            }
-        }
-
-        public bool ISO8601DateFormat
-        {
-            get { return _iso8601DateFormat; }
-            set { _iso8601DateFormat = value; }
-        }
-
-        internal JavaScriptTypeResolver TypeResolver
-        {
-            get { return _typeResolver; }
+                this.RegisterConverters(configuration.Converters, configuration.PrimitiveConverters);

```

Comment:
While we're poking at this class, we might as well fix these. Wrap in curly-braces